x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java

```diff
@@ -259,6 +255,10 @@ public Watcher(final Settings settings) {
                                                NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry,
                                                IndexNameExpressionResolver expressionResolver,
                                                Supplier<RepositoriesService> repositoriesServiceSupplier) {
+        if (enabled && transportClient == false) {
+            validAutoCreateIndex(settings, logger, clusterService);
+        }
```

Comment:
Just double checking: This doesn't result in different behavior, right? Just makes the failure occur (if it occurs) at a different time during node startup.

(also, my assumption is that this moved because at the time the constructor is called we don't have a cluster state yet, correct?)

Reply:
correct, same behavior and moved to allow access the the cluster state.